### ISSUE TITLE
MA0182 is not reported if internal class accessed field

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#965]
### Version of the Meziantou.Analyzer NuGet package

2.0.276

### Rule Identifier

MA0182

### Target Framework

.NET 9.0

### C# Language version

C# 14

### Description

MA0182 is not reported if the internal class accesses (gets or sets) a private field.

* MA0182 is reported:
```c#
internal sealed class NotUsed
{
    private readonly string _name;

    public NotUsed(string name)
    {
    }

    public string X { get; } = "X";

    public string? Whatever()
    {
        return null;
    }
}
```

* MA0182 is **NOT** reported:
```c#
internal sealed class NotUsed
{
    private readonly string _name;

    public NotUsed(string name)
    {
        _name = name;
    }

    public string X { get; } = "X";

    public string? Whatever()
    {
        return _name;
    }
}
```

If you comment out the assignment of **_name** in the ctor and change the **Whatever()** method to return **null**, then **MA0182** will effectively be reported.

**Comment by meziantou** [comment]
Should be fixed by d7ce9d306ea684b7363118a3666325816750f43e
